### PR TITLE
ODATA-1609

### PR DESCRIPTION
**Pull request by HeikoTheissen**
https://issues.oasis-open.org/browse/ODATA-1609

### CODE REVIEW

File: odata-protocol/11.4 Data Modification.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ must not violate the integrity of the data.
 The client may request whether content be returned from a Create,
 Update, or Delete request, or the invocation of an Action, by specifying
 the [`return` Prefer header](#Preferencereturnrepresentationandreturnminimal).
+A [success response](#SuccessResponses) indicates that data have been modified,
+even if the requested content could not be returned.
```
Comment (ralfhandl) [inline odata-protocol/11.4 Data Modification.md:14]:
```suggestion
independent of whether the requested content could be returned.
```

File: odata-protocol/11.4 Data Modification.md [hunk 2]
```diff
@@ -185,18 +187,19 @@ Properties with a defined default value, nullable properties, and
...
-header](#HeaderLocation) that contains the edit URL or read URL of the
-created entity.
-
-Upon successful completion the service MUST respond with either
+Upon successful creation of the entity, the service MUST respond with either
 [`201 Created`](#ResponseCode201Created) and a representation of the
 created entity, or [`204 No Content`](#ResponseCode204NoContent) if the
-request included a [Prefer
-header](#Preferencereturnrepresentationandreturnminimal) with a value of
+request included a [Prefer header](#Preferencereturnrepresentationandreturnminimal)
```
Comment (ralfhandl) [inline odata-protocol/11.4 Data Modification.md:193]:
```suggestion
request included a [`Prefer` header](#Preferencereturnrepresentationandreturnminimal)
```

**Comment by HeikoTheissen** [comment]
Does this change merit mention in section 1.1?

File: odata-protocol/11.4 Data Modification.md [hunk 3]
```diff
@@ -185,18 +187,19 @@ Properties with a defined default value, nullable properties, and
...
+with a value of
 [`return=minimal`](#Preferencereturnrepresentationandreturnminimal) and did not
 include the system query options [`$select`](#SystemQueryOptionselect)
-and [`$expand`](#SystemQueryOptionexpand).
+and [`$expand`](#SystemQueryOptionexpand), or if a representation of the created
+entity could not be constructed. In either case, the response MUST contain a
+[`Location` header](#HeaderLocation) that contains the edit URL or read URL of the
+created entity. If the service could not even construct the `Location` header
+of the created entity, it SHOULD still respond with `204 No Content`, so the client
+can determine the created entity with a [query collection](#QueryingCollections).
```
Comment (ralfhandl) [inline odata-protocol/11.4 Data Modification.md:202]:
```suggestion
can determine the created entity with a [collection query](#QueryingCollections).
```

**Review by ralfhandl — COMMENTED** [review]
Minor nits, see proposals

**Comment by ralfhandl** [comment]
> Does this change deserve mention in section 1.1?

I'd prefer it to be mentioned as we modified/relativized a MUST requirement.

File: odata-protocol/1 Introduction.md [hunk 4]
```diff
@@ -24,8 +24,11 @@ resource representations that are exchanged using OData.
 
 ## ##subsec Changes from Earlier Versions
 
-<!-- TODO -->
-<!-- Describe significant changes from previous differently-numbered Versions, not changes between stages of the current Version -->
+Section | Feature / Change | Issue
+--------|------------------|------
+[Section ##DataModification]| 
+Response code 2xx after successful data modification if requested response could not be constructed|
```
Comment (ralfhandl) [inline odata-protocol/1 Introduction.md:30]:
```suggestion
Response code `204 No Content` after successful data modification if requested response could not be constructed| 
```

File: odata-protocol/11.4 Data Modification.md [hunk 5]
```diff
@@ -9,9 +9,9 @@ must not violate the integrity of the data.
 
 The client may request whether content be returned from a Create,
 Update, or Delete request, or the invocation of an Action, by specifying
-the [`return` Prefer header](#Preferencereturnrepresentationandreturnminimal).
+the [`Prefer: return` header](#Preferencereturnrepresentationandreturnminimal).
```
Comment (ralfhandl) [inline odata-protocol/11.4 Data Modification.md:12]:
```suggestion
the [`return`](#Preferencereturnrepresentationandreturnminimal) preference.
```
Align with most common reference format

File: odata-protocol/11.4 Data Modification.md [hunk 6]
```diff
@@ -190,7 +190,7 @@ default value, null, or an empty collection, respectively.
 Upon successful creation of the entity, the service MUST respond with either
 [`201 Created`](#ResponseCode201Created) and a representation of the
 created entity, or [`204 No Content`](#ResponseCode204NoContent) if the
-request included a [Prefer header](#Preferencereturnrepresentationandreturnminimal)
+request included a [`Prefer` header](#Preferencereturnrepresentationandreturnminimal)
```
Comment (ralfhandl) [inline odata-protocol/11.4 Data Modification.md:193]:
```suggestion
request included a [`Prefer`](#Preferencereturnrepresentationandreturnminimal) header
```

**Review by ralfhandl — COMMENTED** [review]
Sorry, more nits

**Review by ralfhandl — APPROVED** [review]

**Review by ralfhandl — DISMISSED** [review]

**Review by ralfhandl — COMMENTED** [review]
Minor nit

**Comment by HeikoTheissen** [comment]
> Minor nit

Where is the nit?

File: odata-protocol/8 Header Fields.md [hunk 7]
```diff
@@ -932,7 +932,7 @@ In this case, the response body MUST be empty.
 
 As defined in [RFC7231](#rfc7231), a [Data Modification
 Request](#DataModification) that responds with
-`204 No Content MAY `include an `ETag` header with a value reflecting
+`204 No Content` MAY include an `ETag` header with a value reflecting
```
Comment (ralfhandl) [inline odata-protocol/8 Header Fields.md]:
```suggestion
`204 No Content` MAY include an [`ETag`](#HeaderETag) header with a value reflecting
```

**Review by ralfhandl — COMMENTED** [review]
Here

File: odata-protocol/8 Header Fields.md [hunk 8]
```diff
@@ -932,7 +932,7 @@ In this case, the response body MUST be empty.
 
 As defined in [RFC7231](#rfc7231), a [Data Modification
 Request](#DataModification) that responds with
-`204 No Content` MAY include an `ETag` header with a value reflecting
+`204 No Content MAY `include an `ETag` header with a value reflecting
```
Comment (ralfhandl) [inline odata-protocol/8 Header Fields.md]:
```suggestion
`204 No Content MAY `include an [`ETag`](#HeaderETag) header with a value reflecting
```

Comment (ralfhandl) [reply odata-protocol/8 Header Fields.md]: Make `ETag` a link

**Review by ralfhandl — CHANGES_REQUESTED** [review]

**Review by ralfhandl — APPROVED** [review]

**Review by ralfhandl — DISMISSED** [review]
Thanks!

**Review by ralfhandl — APPROVED** [review]

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**